initrd/bin/kexec-boot

```diff
@@ -1,13 +1,89 @@
 #!/bin/sh
 # Launches kexec from saved configuration entries
-set -e -o pipefail
-. /tmp/config
-. /etc/functions
+
+usage() {
+	cat <<End-of-message
+Usage
+=====
+
+Launches kexec from saved configuration entries
+	
+$0 -b /boot -e 'kexec params|...|...' [<options>]
+	
+Options
+-------
+
+### -a <additional confiurations>
+
+Anything provided to the -a argument is appended to the kernel command line (usually comes from parsing the grub, syslinux, BLS files). Ex.
+
+	-a 'intel_iommu=on intel_iommu=igfx_off'
+
+### -b <boot path>
+
+path to boot files.  e.g.  -b /boot
+
+### -e <kexec params>
+
+used to prepare parameters passed to kexec system call for booting operating systems on disk.  Fields are pipe separated. 
+
+	"<description>|[elf,xen,multiboot]|kernel <path>|initrd <path>|<parameters>"
+
+First field is ignored.  The second field is the type of booting.  See examples below.
+
+The other fields are kernel, initrd, and parameters to pass to the linux kernel.
+
+EXAMPLE 1: ELF
+
+boot a linux kernel and specify the root partition
+
+	-e "my custom boot|elf|kernel /vmlinuz|initrd /initrd.img|append root=/dev/sda1"
+
+EXAMPLE 2: XEN
+
+boot Qubes, with Xen hypervisor
+
+	-e "xen boot|xen|module /vmlinuz|module /vmlinuz-4.4.67-13.x86_64 placeholder root=/dev/mapper/root ro|module /initramfs-4.4.67-13.x86_64.img"
+
+EXAMPLE 3: MULTIBOOT
+
+boot Qubes, with Xen hypervisor and LUKS encrypted root partition
+
+	-e "Qubes, with Xen hypervisor|multiboot|kernel /xen-4.6.5.gz placeholder |module /vmlinuz-4.4.67-13.pvops.qubes.x86_64 placeholder root=/dev/mapper/luks-UUID ro rd.qubes.hide_all_usb|module /initramfs-4.4.67-13.pvops.qubes.x86_64.img"
+
+Currently, any multiboot entry is interpreted as a Xen-variant and kexec-boot overrides the arguments to the multiboot kernel with custom arguments. 
+
+### -f
+
+some sort of debug mode which prints information about files and sets dryrun on
+
+### -h
+
+show help
+
+### -i
+
+Use this parameter to ignore the verification of boot hashes and rollback counter.  This will boot a potentially compromised system.
+
+### -o <initrd>
+
+use the specified initrd for booting
+
+### -r <configurations to remove>
+
+Anything provided to the -r argument is removed from the kernel command line. Ex.
+
+	-r 'quiet'
+
```

Comment:
@jtmoree-github-com 
[kernel option remove statements normally come from board configuration](https://github.com/osresearch/heads/blob/0d5b3d75e7b84ca85ff252fe5fe23aea990da435/initrd/bin/kexec-boot#L32-L33) to pass board specific tweaks, needed for that board from OS perspective.

@jtmoree-github-com Something like that. :)

Reply:
I have reworded this to follow the suggestion